frontend/app.tsx

```diff
@@ -255,21 +266,35 @@ function reducer(
       if (action.filters.equals(state.filters)) {
         return state;
       }
-      const allIssues = [...state.allIssuesMap.values()];
+      // TODO (mlaw): re-hydrate stream from past events
       return {
         ...state,
-        viewIssueCount: countViewIssues(allIssues),
+        views: {
+          issueCount: issueCountView(state.issueSource, filters.viewFilter),
+          filteredIssues: filteredIssuesView(
+            state.issueSource,
+            issueOrder,
+            filters.issuesFilter
+          ),
+        },
         filters: action.filters,
-        filteredIssues: filterAndSort(allIssues),
       };
     }
     case "setIssueOrder": {
       if (action.issueOrder === state.issueOrder) {
         return state;
       }
+
       return {
         ...state,
-        filteredIssues: sortBy(state.filteredIssues, orderIteratee),
+        views: {
```

Comment:
I think this is the same code as in the `setFilters` block. Perhaps this higher level reducer code can now be simplified? We want this to serve as a good example of how to do this kind of thing, so if this no longer makes sense, let's fix.